src/js_native_api_v8.h

```diff
@@ -51,8 +51,13 @@ class Finalizer;
 }  // end of namespace v8impl
 
 struct napi_env__ {
-  explicit napi_env__(v8::Local<v8::Context> context)
-      : isolate(context->GetIsolate()), context_persistent(isolate, context) {
+  explicit napi_env__(v8::Local<v8::Context> context,
+                      int32_t module_api_version)
+      : isolate(context->GetIsolate()),
+        context_persistent(isolate, context),
+        module_api_version(module_api_version != 0
```

Comment:
IIUC, if the callsite of the callback `GetNapiAddonGetApiVersionCallback` set the `module_api_version` defaults to `NODE_API_DEFAULT_MODULE_API_VERSION` instead of `0`, this conditional branch can be removed.

Reply:
I would prefer doing such checks closer to the use of the variable because we cannot know all possible callers in future. 
Anyway, I followed the suggestion and removed the check.